spec/es_spec_helper.rb

```diff
@@ -34,14 +34,26 @@ def get_client
 RSpec.configure do |config|
   config.include ESHelper
 
-  config.before(:suite, :integration => true) do
-    @docker = Longshoreman.new("#{CONTAINER_IMAGE}:#{CONTAINER_TAG}", CONTAINER_NAME)
-    # TODO(talevy): detect when ES is up and ready instead of sleeping
-    # an arbitrary amount
-    sleep 10
+  config.before(:all, :integration => true) do
+    # check if container exists already
+    begin
+      ls = Longshoreman::new
+      ls.container.get(CONTAINER_NAME)
+    rescue Docker::Error::NotFoundError
+      Longshoreman.new("#{CONTAINER_IMAGE}:#{CONTAINER_TAG}", CONTAINER_NAME)
+      # TODO(talevy): verify ES is running instead of static timeout
+      sleep 10
+    end
   end
 
-  config.after(:suite, :integration => true) do
-    @docker.cleanup
+  config.after(:suite) do
```

Comment:
we want to do a final cleanup after all `:integration` runs... but we don't want to clean up before the last block... so we must cleanup after the suite has run. so this is a final blind check to see if the ES docker container is running and needs to be cleaned up. If no container can be found and/or docker is not running on the system, we do nothing